app/controllers/application_controller.rb

```diff
@@ -122,6 +122,12 @@ def grammatical_gender
   # use :mobile view for mobile and :html for everything else
   # (except if explicitly specified, e.g. :json, :xml)
   def mobile_switch
+    # mobile_fu's is_mobile_device? wasn't working here for some reason...
+    # it may have been just because of the test env.
+    if session[:mobile_view].nil? && request.env['HTTP_USER_AGENT'].try(:match, /mobile/i)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.